tfhe/src/integer/server_key/radix_parallel/div_mod.rs

```diff
@@ -0,0 +1,496 @@
+            // latency of this function won't be impacted (compared to doing it right now).
+            let mut is_remainder_greater_or_eq_than_divisor;
+            if self.key.message_modulus.0 < 3 {
+                let merged_cmp = self.key.unchecked_apply_lookup_table_bivariate(
+                    &trivial_blocks_are_zero,
+                    &non_trivial_blocks_are_ge.blocks[0],
+                    &merge_two_cmp_lut,
+                );
+
+                interesting_divisor.blocks.par_iter_mut().for_each(|block| {
+                    self.key.unchecked_apply_lookup_table_bivariate_assign(
+                        block,
+                        &merged_cmp,
+                        &zeroer_lut_for_merged_cmp,
+                    );
+                });
+                is_remainder_greater_or_eq_than_divisor = Some(merged_cmp)
+            } else {
+                let summed_cmp = self.key.unchecked_add(
+                    &trivial_blocks_are_zero,
+                    &non_trivial_blocks_are_ge.blocks[0],
+                );
+                interesting_divisor.blocks.par_iter_mut().for_each(|block| {
+                    self.key.unchecked_apply_lookup_table_bivariate_assign(
+                        block,
+                        &summed_cmp,
+                        &zeroer_lut_summed_cmp,
+                    );
+                });
+                is_remainder_greater_or_eq_than_divisor = None;
+            }
+
+            rayon::join(
+                || {
+                    self.sub_assign_parallelized(&mut interesting_remainder, &interesting_divisor);
+                    // Copy back into the real remainder
+                    remainder.blocks[..first_trivial_block]
+                        .iter_mut()
+                        .zip(interesting_remainder.blocks.iter())
+                        .for_each(|(remainder_block, new_value)| {
+                            remainder_block.copy_from(new_value);
+                        });
+                },
+                || {
+                    // This is the place where we merge the two cmp blocks
+                    // if it was not done earlier.
+                    let merged_cmp =
+                        is_remainder_greater_or_eq_than_divisor.get_or_insert_with(|| {
+                            self.key.unchecked_apply_lookup_table_bivariate(
+                                &trivial_blocks_are_zero,
+                                &non_trivial_blocks_are_ge.blocks[0],
+                                &merge_two_cmp_lut,
+                            )
+                        });
+                    if pos_in_block != 0 {
+                        self.key
+                            .unchecked_scalar_left_shift_assign(merged_cmp, pos_in_block as u8);
+                    }
+                    self.key
+                        .unchecked_add_assign(&mut quotient.blocks[block_of_bit], merged_cmp);
```

Comment:
isn't there something missing when jumping from a block to another maybe ?

Reply:
what do you mean ?

We shift the value by `pos_in_block` and add it to the block at `block_of_bit`

Reply:
Yes, here we are doing `Q(i) := 1`
where instead of one we have a homomorphic shortint block resulting from `R>=D`.
and Q is our typical RadixCiphertext, `i` the index of the loop that goes from `nb_bits-1 to 0`

Reply:
The scalar_left shift directly calls scalar_mul with `1 << scalar` as the multiplier.

Maybe that if can be moved to scalar_left_shift (or even scalar_mul)

Reply:
yes as it would shortcut for trivial cases, wdyt ?